pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestorage
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/lib/controller"
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/filestorage"
+	"github.com/oracle/oci-go-sdk/identity"
+
+	"github.com/oracle/oci-volume-provisioner/pkg/oci/client"
+	"github.com/oracle/oci-volume-provisioner/pkg/provisioner/plugin"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ociVolumeID            = "volume.beta.kubernetes.io/oci-volume-id"
+	ociExportID            = "volume.beta.kubernetes.io/oci-export-id"
+	volumePrefixEnvVarName = "OCI_VOLUME_NAME_PREFIX"
+	fsType                 = "fsType"
+	subnetID               = "subnetId"
+	mntTargetID            = "mntTargetId"
+)
+
+// filesystemProvisioner is the internal provisioner for OCI filesystem volumes
+type filesystemProvisioner struct {
+	client client.ProvisionerClient
+}
+
+var _ plugin.ProvisionerPlugin = &filesystemProvisioner{}
+
+// NewFilesystemProvisioner creates a new file system provisioner that creates
+// filsystems using OCI file system service.
+func NewFilesystemProvisioner(client client.ProvisionerClient) plugin.ProvisionerPlugin {
+	rand.Seed(time.Now().Unix())
+	return &filesystemProvisioner{
+		client: client,
+	}
+}
+
+// getMountTargetFromID retrieves mountTarget from given mountTargetID
+func getMountTargetFromID(ctx context.Context, mountTargetID string, fileStorageClient client.FileStorage) (*filestorage.MountTarget, error) {
+	responseMnt, err := fileStorageClient.GetMountTarget(ctx, filestorage.GetMountTargetRequest{
+		MountTargetId: common.String(mountTargetID),
+	})
+	if err != nil {
+		glog.Errorf("Failed to retrieve mount point mountTargetId=%q: %v", mountTargetID, err)
+		return nil, err
+	}
+	return &responseMnt.MountTarget, nil
+}
+
+func getOrCreateMountTarget(ctx context.Context, availabilityDomain *string, subnetID string, compartmentID string, fileStorageClient client.FileStorage) (id *filestorage.MountTarget, err error) {
```

Comment:
No need to name the return params here (you're not using naked returns and nor would it be a particularly good idea to do so).

Reply:
Same re private method on the struct

Reply:
Which one?